### DISCUSSION TITLE
MIDI snapshot saving

### DISCUSSION THREAD

**Discussion by EE1977** [rerdavies/pipedal#260] [Ideas]
Hey Robin!

I hope you're doing well! I have a fun idea I wanted to share with you. Wouldn't it be great to have an option to save snapshots directly from MIDI? For instance, I could use my MIDI foot controller to pick the snapshot I want, make a few tweaks right from the controller, and then save my updated snapshot just by long-pressing the snapshot button switch. What do you think? Can't wait to hear your thoughts!

**Comment by EE1977** [comment]
In fact, I think it would be super helpful to have a long press (or maybe even a double press) MIDI function as a universal option in Pipedal. What do you think?

**Comment by rerdavies** [comment]
Possible, now that PiPedal (imminently) supports LV2 trigger controls properly. I'd have to look at the code, and give some thought as to what the UI would be like. 

Save Preset is also an interesting idea.  Promoted to a feature request.

**Reply by EE1977** [reply]
Great news Robin!! Wish you a happy new year!!
